api/src/test/java/uk/gov/dhsc/htbhf/claimant/reporting/ClaimantCategoryCalculatorTest.java

```diff
@@ -0,0 +1,138 @@
+        LocalDate now = LocalDate.now();
+        List<LocalDate> datesOfBirthOfChildren = singletonList(LocalDate.now().minusYears(2));
+        Claimant claimant = aValidClaimantBuilder()
+                .expectedDeliveryDate(expectedDeliveryDate)
+                .dateOfBirth(LocalDate.now().minusYears(20))
+                .build();
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(true);
+
+        ClaimantCategory claimantCategory = claimantCategoryCalculator.determineClaimantCategory(claimant, datesOfBirthOfChildren, now);
+
+        assertThat(claimantCategory).isEqualTo(PREGNANT_WITH_CHILDREN);
+        ArgumentCaptor<LocalDate> argumentCaptor = ArgumentCaptor.forClass(LocalDate.class);
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(eq(expectedDeliveryDate), argumentCaptor.capture());
+        assertThat(argumentCaptor.getValue()).isAfterOrEqualTo(now);
+    }
+
+    @Test
+    void shouldMatchClaimantToNotPregnantWithChildren() {
+        LocalDate expectedDeliveryDate = null;
+        LocalDate now = LocalDate.now();
+        List<LocalDate> datesOfBirthOfChildren = singletonList(LocalDate.now().minusYears(2));
+        Claimant claimant = aValidClaimantBuilder()
+                .expectedDeliveryDate(expectedDeliveryDate)
+                .dateOfBirth(LocalDate.now().minusYears(20))
+                .build();
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+
+        ClaimantCategory claimantCategory = claimantCategoryCalculator.determineClaimantCategory(claimant, datesOfBirthOfChildren, now);
+
+        assertThat(claimantCategory).isEqualTo(NOT_PREGNANT_WITH_CHILDREN);
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(expectedDeliveryDate, now);
+    }
+
+    @Test
+    void shouldMatchClaimantToNotPregnantWithNoChildren() {
+        LocalDate expectedDeliveryDate = null;
+        LocalDate now = LocalDate.now();
+        List<LocalDate> datesOfBirthOfChildren = emptyList();
+        Claimant claimant = aValidClaimantBuilder()
+                .expectedDeliveryDate(expectedDeliveryDate)
+                .dateOfBirth(LocalDate.now().minusYears(20))
+                .build();
+        given(pregnancyEntitlementCalculator.isEntitledToVoucher(any(), any())).willReturn(false);
+
+        ClaimantCategory claimantCategory = claimantCategoryCalculator.determineClaimantCategory(claimant, datesOfBirthOfChildren, now);
+
+        assertThat(claimantCategory).isEqualTo(NOT_PREGNANT_WITH_NO_CHILDREN);
+        verify(pregnancyEntitlementCalculator).isEntitledToVoucher(expectedDeliveryDate, now);
+    }
+
+    @Test
+    void shouldMatchClaimantToNotPregnantWithNoChildrenWhenChildrenOver4() {
+        LocalDate expectedDeliveryDate = null;
+        LocalDate now = LocalDate.now();
+        List<LocalDate> datesOfBirthOfChildren = singletonList(LocalDate.now().minusYears(4));
```

Comment:
Is this specifically testing the 4 year boundary case, if so, it should be explicit in the test name, otherwise I'd make their DOB a bit older so its definitely over 4 and not just 4.